pkg/latte/node_do_update_authenticator.go

```diff
@@ -22,7 +23,14 @@ func (n *NodeDoUpdateAuthenticator) Kind() string {
 func (n *NodeDoUpdateAuthenticator) GetEffects(ctx context.Context, deps *workflow.Dependencies, workflows workflow.Workflows) (effs []workflow.Effect, err error) {
 	return []workflow.Effect{
 		workflow.RunEffect(func(ctx context.Context, deps *workflow.Dependencies) error {
-			return deps.Authenticators.Update(n.Authenticator)
+			if n.Authenticator.Password != nil {
+				return deps.Authenticators.UpdatePassword(n.Authenticator, &password.UpdatePasswordOptions{
+					PasswordHash:   n.Authenticator.Password.PasswordHash,
+					SetExpireAfter: true,
+				})
+			} else {
+				return deps.Authenticators.Update(n.Authenticator)
+			}
```

Comment:
This is incorrect. in pkg/latte/node_change_password.go, there is a call to WithSpec. I said replace WithSpec with UpdatePassword, when the authenticator passed to WithSpec is a password authenticator.

I expect to list out all WithSpec calls, and find out which one should switch to UpdatePassword.

Please read my original commend again. https://github.com/authgear/authgear-server/pull/4511#discussion_r1699516797